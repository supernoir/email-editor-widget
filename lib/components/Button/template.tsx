import React from 'react'
import { StyledButton } from './styles'

interface ButtonInterface {
  label: string,
  onClick: (evt: React.MouseEvent<Element, MouseEvent>) => void
}

export const Button: React.FC<ButtonInterface> = ({ label, onClick }) => {
  return <StyledButton
    onClick={(evt: React.MouseEvent<Element, MouseEvent>) => onClick(evt)}
  >
    {label}
  </StyledButton>
}