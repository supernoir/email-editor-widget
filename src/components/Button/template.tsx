import React from 'react'
import { StyledButton } from './styles'

interface ButtonInterface {
  label: string,
  onClick: (event: any) => void
}

export const Button: React.FC<ButtonInterface> = ({ label, onClick }) => {
  return <StyledButton
    onClick={onClick}
  >
    {label}
  </StyledButton>
}