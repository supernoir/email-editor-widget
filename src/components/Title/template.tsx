import React from 'react'
import { StyledTitle, StyledHighlight } from './styles'

interface TitleProps {
  shareableItem: string
}

export const Title: React.FC<TitleProps> = ({ shareableItem }) => {
  const displayTitle = (item: string) => {
    return (<span>
      Share <StyledHighlight>{item}</StyledHighlight> with others
    </span>)
  }
  return <StyledTitle>{displayTitle(shareableItem)}</StyledTitle>
}