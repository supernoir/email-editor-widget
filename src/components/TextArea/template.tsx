import React, { useState } from 'react'
import { StyledTextAreaView, StyledTextAreaEdit, StyledItem, StyledInvalidItem } from './styles'
import { RemoveIcon } from '../RemoveIcon/template'
import { ShareableTypes } from '../../constants/enums'
import { validateEmailAddress } from '../../util/validity'

interface TextAreaProps {
  itemList: string[],
  removeItemFromList: (item: string) => void
}

enum TextAreaModes {
  edit = "edit",
  view = "view"
}

export const TextArea: React.FC<TextAreaProps> = ({ itemList, removeItemFromList }) => {
  const [mode, setMode] = useState(TextAreaModes.view)

  const handleRemoveItem = (item: string) => {
    console.log(`${item} will be removed`)
    removeItemFromList(item)
  }

  const appendToList = (list: string[], item: string) => {
    return list.push(item)
  }

  const handleKeyPress = (event: { charCode: number }) => {
    if (event.charCode === 13) {
      setMode(TextAreaModes.view)
    }
  }
  const handleMouseClick = () => {
    setMode(TextAreaModes.edit)
  }

  const checkItemValidity = (item: string, type: ShareableTypes) => {
    switch (type) {
      case ShareableTypes.twitterHandle:
      // TODO: handle validity check for twitter handle
      case ShareableTypes.fbUserName:
      // TODO: handle validity check for Facebook username
      default: {
        // Default shareable type is email
        return validateEmailAddress(item)
      }
    }
  }

  const displayItem = (item: string) => {
    if (checkItemValidity(item, ShareableTypes.email)) {
      return <StyledItem key={item}>
        {item} <RemoveIcon handleClick={() => handleRemoveItem(item)} />
      </StyledItem>
    } else {
      return <StyledInvalidItem>
        {item}
      </StyledInvalidItem>
    }
  }

  return mode && mode === TextAreaModes.view
    ? (<StyledTextAreaView onClick={handleMouseClick}>
      {itemList && itemList.map((item) => {
        return displayItem(item)
      })}
    </StyledTextAreaView>)
    : (<StyledTextAreaEdit onKeyPress={handleKeyPress} value={itemList && itemList.map((item) => {
      return item
    })} />
    )
}