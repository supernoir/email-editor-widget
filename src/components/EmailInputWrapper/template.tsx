import React from 'react'
import { StyledEmailInputWrapper } from './styles'
import { Title } from '../Title/template'
import { TextArea } from '../TextArea/template'
import { Item, ItemList } from '../../types'

interface EmailInputWrapperProps {
  shareableItem: string,
  itemList: ItemList,
  removeItemFromList: (item: Item) => void,
  addToItemsList: (items: ItemList) => void
}

export const EmailInputWrapper: React.FC<EmailInputWrapperProps> = ({ shareableItem, itemList, removeItemFromList, addToItemsList }) => {
  return <StyledEmailInputWrapper>
    <Title shareableItem={shareableItem} />
    <TextArea
      itemList={itemList}
      removeItemFromList={removeItemFromList}
      addToItemsList={addToItemsList}
    />
  </StyledEmailInputWrapper>
}