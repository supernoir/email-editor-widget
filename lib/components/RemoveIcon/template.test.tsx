import React from 'react';
import { RemoveIcon } from './template';
import Renderer from 'react-test-renderer';

const mockHandleClick = jest.fn()

describe('RemoveIcon', () => {
  test('should match the snapshot', () => {
    const component = Renderer.create(<RemoveIcon
      handleClick={mockHandleClick}
    />);
    expect(component).toMatchSnapshot();
  });
});