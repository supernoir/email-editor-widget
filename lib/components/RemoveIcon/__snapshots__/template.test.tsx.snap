// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveIcon should match the snapshot 1`] = `
<svg
  fill="none"
  height="8"
  onClick={[MockFunction]}
  viewBox="0 0 8 8"
  width="8"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M8 0.8L7.2 0L4 3.2L0.8 0L0 0.8L3.2 4L0 7.2L0.8 8L4 4.8L7.2 8L8 7.2L4.8 4L8 0.8Z"
    fill="#050038"
    fillRule="evenodd"
  />
</svg>
`;
